aten/src/ATen/native/AutogradLab.cpp

```diff
@@ -0,0 +1,19 @@
+#include <ATen/ATen.h>
+#include <ATen/core/DimVector.h>
+#include <c10/util/SmallBuffer.h>
+
+#include <vector>
+
+namespace at {
+namespace native {
+
+std::tuple<Tensor, Tensor> attn(const Tensor& q, const Tensor& k, const Tensor& v) {
+    TORCH_CHECK(!q.is_complex() && q.dtype() != kBFloat16 && q.dtype() != at::kHalf,
+            "torch.attn: not implemented for ", q.dtype());
+
```

Comment:
There should probably be error checking on the shape of `q, k, v`, here. If `attn` is a user facing operator, then users might be confused if an error message gets raised in `matmul`: a good error message would say something like "attn(q, k, v): q.shape (...) and k.shape (...) are incompatible"